tests/unit/test_fm.py

```diff
@@ -94,3 +108,282 @@ def test_all_hyperparameters(sagemaker_session):
+
+
+def test_linear_init_sigma_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(linear_init_sigma='invalid', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_linear_init_sigma_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(linear_init_sigma=-1, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_linear_init_value_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(linear_init_value='invalid', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_factors_init_method_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(factors_init_method=0, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_factors_init_method_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(factors_init_method='invalid', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_factors_init_scale_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(factors_init_scale='invalid', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_factors_init_scale_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(factors_init_scale=-1, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_factors_init_sigma_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(factors_init_sigma='invalid', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_factors_init_sigma_validation_fail_value(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(factors_init_sigma=-1, sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+def test_factors_init_value_validation_fail_type(sagemaker_session):
+    with pytest.raises(ValueError):
+        FactorizationMachines(factors_init_value='invalid', sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+
+PREFIX = "prefix"
+FEATURE_DIM = 10
+MINI_BATCH_SIZE = 200
+
+
+@patch("sagemaker.amazon.amazon_estimator.AmazonAlgorithmEstimatorBase.fit")
+def test_call_fit(base_fit, sagemaker_session):
+    fm = FactorizationMachines(base_job_name="fm", sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
+
+    data = RecordSet("s3://{}/{}".format(BUCKET_NAME, PREFIX), num_records=1, feature_dim=FEATURE_DIM, channel='train')
+
+    fm.fit(data, MINI_BATCH_SIZE)
+
+    base_fit.assert_called_once()
+    assert len(base_fit.call_args[0]) == 2
+    assert base_fit.call_args[0][0] == data
+    assert base_fit.call_args[0][1] == MINI_BATCH_SIZE
+
+
+def test_call_fit_none_mini_batch_size(sagemaker_session):
```

Comment:
What is this test asserting on? And does the base fit need to be patched? (Same question about patch applies to the tests below as well)